Frends.AzureEventHub.Receive/Frends.AzureEventHub.Receive/AzureEventHub.cs

```diff
@@ -0,0 +1,139 @@
+    /// <returns>Object { bool Success, List&lt;dynamic&gt; Data }</returns>
+    public static async Task<Result> Receive([PropertyTab] Consumer consumer, [PropertyTab] Checkpoint checkpoint, [PropertyTab] Options options, CancellationToken cancellationToken)
+    {
+        if (options.MaxEvents.Equals(0) && options.MaximumWaitTimeInMinutes.Equals(0))
+            throw new Exception("Both Options.MaxEvents and Options.MaximumWaitTimeInMinutes cannot be unlimited.");
+
+        var results = new List<dynamic>();
+        var stopProcessing = false;
+        EventProcessorClient processorClient = null;
+        var timeOut = options.MaximumWaitTimeInMinutes > 0 ? DateTime.UtcNow.AddMinutes(options.MaximumWaitTimeInMinutes) : DateTime.UtcNow;
+        var cancellationTokenSource = new CancellationTokenSource();
+
+
+        try
+        {
+            var checkpointStorageClient = CreateBlobContainerClient(checkpoint);
+
+            if (checkpoint.CreateContainer && checkpoint.AuthenticationMethod != AuthenticationMethod.SASToken)
+                await checkpointStorageClient.CreateIfNotExistsAsync(PublicAccessType.None, null, null, cancellationToken);
+
+            processorClient = CreateEventProcessorClient(consumer, checkpointStorageClient);
+
+            processorClient.ProcessEventAsync += async (args) =>
+            {
+                if (options.MaximumWaitTimeInMinutes > 0 && timeOut <= DateTime.UtcNow)
+                    stopProcessing = true;
+                else if (options.MaxEvents > 0 && results.Count >= options.MaxEvents)
+                    stopProcessing = true;
+                else if (args.Data != null && !stopProcessing)
+                {
+                    var result = Encoding.UTF8.GetString(args.Data.Body.ToArray());
+                    results.Add(result);
+                    await args.UpdateCheckpointAsync(cancellationToken);
+                }
+            };
+
+            processorClient.ProcessErrorAsync += async (args) =>
+            {
+                var ex = @$"Partition {args.PartitionId}, Exception: {args.Exception}";
+
+                if (options.ExceptionHandler is ExceptionHandlers.Info)
+                {
+                    results.Add(ex);
+                    await Task.CompletedTask;
+                }
+                else
+                    throw new Exception(ex);
+            };
+
+            await processorClient.StartProcessingAsync(cancellationToken);
+
+            while (!stopProcessing)
+                await Task.Delay(TimeSpan.FromSeconds(options.Delay), cancellationToken);
+
+            //if (processorClient.IsRunning)
+            //    await processorClient.StopProcessingAsync(cancellationToken);
+
+        }
+        catch (Exception ex)
+        {
+            if (options.ExceptionHandler is ExceptionHandlers.Throw)
+                throw new Exception($@"{ex}");
+            else
+            {
+                results.Add(@$"An exception occured: {ex}");
```

Comment:
Same thing in here.

Reply:
Removed @. I'm using both of those because it doesn't affect memory usage or the final result and it's just more user-friendly to not to think about when to use the combo.